src/fhir-products/fhir-products.controller.ts

```diff
@@ -0,0 +1,43 @@
+import {
+  Controller,
+  Get,
+  Query,
+  BadGatewayException,
+  NotFoundException,
+  Param,
+} from '@nestjs/common';
+import { R4 } from '@ahryman40k/ts-fhir-types';
+import { ProductsService } from 'src/products/products.service';
+import { ProductsQuery } from 'src/common/interfaces/products-query.interface';
+import {
+  formatProductsToFhir,
+  getSystems,
+} from 'src/common/utils/fhir/formatter';
+import { ProductDto } from 'src/common/dtos/product.dto';
+
+@Controller('fhir')
+export class FhirProductsController {
+  constructor(private service: ProductsService) {}
+
+  @Get()
+  async getFhir(@Query() query: ProductsQuery): Promise<R4.IConceptMap> {
+    try {
+      const { products } = await this.service.findAll(query);
+
+      return formatProductsToFhir(products, getSystems(products));
+    } catch (error) {
+      throw new BadGatewayException(error.message);
+    }
+  }
+
+  @Get(':id')
+  async findOne(@Param('id') id: string): Promise<R4.IConceptMap> {
+    try {
+      const product = await this.service.findOne(id, {});
+      const products = [product] as ProductDto[];
```

Comment:
This seems like you need a function to format one product then you can reuse that for >= 1 products.

Reply:
alright will rework on that.